subgraphs/friend-tech/src/common/metrics.ts

```diff
@@ -0,0 +1,307 @@
+import { ethereum, BigInt, Address } from "@graphprotocol/graph-ts";
+
+import {
+  getOrCreateProtocol,
+  getOrCreateEthToken,
+  getOrCreateAccount,
+  getOrCreateTrader,
+  getOrCreateSubject,
+  getOrCreateTraderOfType,
+  getOrCreateConnection,
+} from "./getters";
+import { BIGINT_TEN, ETH_DECIMALS, INT_ONE } from "./constants";
+import { addToArrayAtIndex } from "./utils";
+
+export function updateTVL(
+  amountETH: BigInt,
+  isBuy: boolean,
+  event: ethereum.Event
+): void {
+  const protocol = getOrCreateProtocol();
+  const eth = getOrCreateEthToken(event);
+
+  const amountUSD = amountETH
+    .toBigDecimal()
+    .div(BIGINT_TEN.pow(ETH_DECIMALS as u8).toBigDecimal())
+    .times(eth.lastPriceUSD!);
+
+  if (isBuy) {
+    protocol.totalValueLockedETH = protocol.totalValueLockedETH.plus(amountETH);
+    protocol.totalValueLockedUSD = protocol.totalValueLockedUSD.plus(amountUSD);
```

Comment:
Thoughts on recalculating the USD TVL with the new ETH amounts? That is usually how we do it, but I am not 100% sure if that does/doesn't make sense for friend tech